collect_app/src/main/java/org/odk/collect/android/widgets/RatingWidget.java

```diff
@@ -14,116 +14,92 @@
+        ratingBar.setOnRatingBarChangeListener((ratingBar, rating, fromUser) -> widgetValueChanged());
+        ratingBar.setEnabled(!prompt.isReadOnly());
 
-        for (int column = 0,
-             starId = 0;
-             starId < numberOfStars;
-             column++, starId++) {
-            column = column == columns ? 0 : column;
-
-            ImageButton imageButton = createImageButton(context, numberOfStars, starId);
-            gridLayout.addView(imageButton);
+        if (prompt.getAnswerText() != null) {
+            ratingBar.setRating(Integer.parseInt(prompt.getAnswerText()));
         }
-    }
-
-    private ImageButton createImageButton(Context context, int numberOfStars, int total) {
-        ImageButton imageButton = new ImageButton(context);
-        imageButton.setImageResource(R.drawable.ic_star_border);
-        imageButton.setId(total);
-        imageButton.setPadding(0, 0, 0, 0);
-        imageButton.setBackground(null);
-        imageButton.setEnabled(!getFormEntryPrompt().isReadOnly());
-        imageButton.setOnClickListener(view -> {
-            int position = view.getId();
-            for (int i = 0; i < numberOfStars; i++) {
-                ImageButton button = (ImageButton) gridLayout.getChildAt(i);
-                button.setImageResource(i <= position ? R.drawable.ic_star : R.drawable.ic_star_border);
-            }
-            answer = position + 1;
-
-            widgetValueChanged();
-        });
-        return imageButton;
+        return answerView;
     }
 
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
-        for (int i = 0; i < gridLayout.getChildCount(); i++) {
-            gridLayout.getChildAt(i).setOnLongClickListener(l);
-        }
+        ratingBar.setOnLongClickListener(l);
     }
 
     @Override
     public IAnswerData getAnswer() {
-        return answer != null ? new IntegerData(answer) : null;
+        return ratingBar.getRating() == 0.0F ? null : new IntegerData((int) ratingBar.getRating());
     }
 
     @Override
     public void clearAnswer() {
-        if (answer != null) {
-            for (int i = 0; i < answer; i++) {
-                ((ImageButton) gridLayout.getChildAt(i)).setImageResource(R.drawable.ic_star_border);
-            }
-        }
+        ratingBar.setRating(0.0F);
+    }
 
-        answer = null;
-        widgetValueChanged();
+    protected RatingBar getRatingBar() {
+        return ratingBar;
     }
 
-    private int calculateColumns() {
-        Drawable starDrawable = getResources().getDrawable(R.drawable.ic_star);
+    private int calculateColumns(int widthOfStar) {
         DisplayMetrics dm = getScreenDimensions((Activity) getContext());
-        int widthOfStar = starDrawable.getIntrinsicWidth();
-        return (dm.widthPixels - ASSUMED_TOTAL_MARGIN_AROUND_WIDGET) / widthOfStar;
+        return (dm.widthPixels - ASSUMED_TOTAL_MARGIN_AROUND_WIDGET) / ((int) UiUtils.convertDpToPixel(widthOfStar, getContext()));
```

Comment:
I think you can  use `ScreenUtils.getScreenWidth()` here and get rid of `getScreenDimensions()` at all.